server/lib/job-queue/handlers/move-to-object-storage.ts

```diff
@@ -0,0 +1,138 @@
+
+    const filename = file.filename
+    await storeObject(
+      { filename, path: join(CONFIG.STORAGE.VIDEOS_DIR, file.filename) },
+      CONFIG.OBJECT_STORAGE.VIDEOS
+    )
+
+    file.storage = VideoStorageType.OBJECT_STORAGE
+    file.fileUrl = generateObjectStoreUrl(filename, CONFIG.OBJECT_STORAGE.VIDEOS)
+    await file.save()
+  }
+}
+
+async function moveHLSFiles (video: MVideoWithAllFiles, videoFileId: number) {
+  for (const playlist of video.VideoStreamingPlaylists) {
+    const baseHlsDirectory = join(HLS_STREAMING_PLAYLIST_DIRECTORY, video.uuid)
+
+    for (const file of playlist.VideoFiles) {
+      if (file.storage !== VideoStorageType.LOCAL) continue
+      if (videoFileId !== null && file.id !== videoFileId) continue
+
+      // Resolution playlist
+      const playlistFileName = getHlsResolutionPlaylistFilename(file.filename)
+      await storeObject(
+        {
+          filename: join(playlist.getStringType(), video.uuid, playlistFileName),
+          path: join(baseHlsDirectory, playlistFileName)
+        },
+        CONFIG.OBJECT_STORAGE.STREAMING_PLAYLISTS
+      )
+
+      // Resolution fragmented file
+      const filename = join(playlist.getStringType(), video.uuid, file.filename)
+      await storeObject(
+        {
+          filename,
+          path: join(baseHlsDirectory, file.filename)
+        },
+        CONFIG.OBJECT_STORAGE.STREAMING_PLAYLISTS
+      )
+
+      // Signals that the video file + playlist file were uploaded
+      file.storage = VideoStorageType.OBJECT_STORAGE
+      file.fileUrl = generateObjectStoreUrl(filename, CONFIG.OBJECT_STORAGE.STREAMING_PLAYLISTS)
+      await file.save()
+    }
+  }
+}
+
+async function doAfterLastJob (video: MVideoWithAllFiles) {
+  for (const playlist of video.VideoStreamingPlaylists) {
+    const baseHlsDirectory = join(HLS_STREAMING_PLAYLIST_DIRECTORY, video.uuid)
+    // Master playlist
+    const masterPlaylistFilename = join(playlist.getStringType(), video.uuid, playlist.playlistFilename)
+    await storeObject(
+      {
+        filename: masterPlaylistFilename,
+        path: join(baseHlsDirectory, playlist.playlistFilename)
+      },
+      CONFIG.OBJECT_STORAGE.STREAMING_PLAYLISTS
+    )
+
+    // Sha256 segments file
+    const segmentsFileName = join(playlist.getStringType(), video.uuid, playlist.segmentsSha256Filename)
+    await storeObject(
+      {
+        filename: segmentsFileName,
+        path: join(baseHlsDirectory, playlist.segmentsSha256Filename)
+      },
+      CONFIG.OBJECT_STORAGE.STREAMING_PLAYLISTS
+    )
+
+    playlist.playlistUrl = generateObjectStoreUrl(masterPlaylistFilename, CONFIG.OBJECT_STORAGE.STREAMING_PLAYLISTS)
+    playlist.segmentsSha256Url = generateObjectStoreUrl(segmentsFileName, CONFIG.OBJECT_STORAGE.STREAMING_PLAYLISTS)
+    playlist.storage = VideoStorageType.OBJECT_STORAGE
+    await playlist.save()
+  }
+
+  // Remove files that were "moved"
+  const tasks: Promise<any>[] = []
```

Comment:
Why do you remove files in this function? I think we should remove them just after a `storeObject` success

Reply:
I think the highest resolution transcode is used as input for the other transcodes, so that file is still needed even after it's uploaded to object storage. This function is only run after all the uploades are done so then we don't have that problem

Reply:
I changed this in aa6894eeb, but I'm not sure if it's gonna work due to my above comment